src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/IndexOfAny2Values.cs

```diff
@@ -19,6 +20,10 @@ public IndexOfAny2Values(ReadOnlySpan<T> values)
 
         internal override T[] GetValues() => new[] { _e0, _e1 };
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal override bool ContainsCore(T value) =>
+            value == _e0 || value == _e1;
```

Comment:
Just curious what the perf looks like if this is made branchless with an `|` instead of an `||`.  I don't know if that's a good tradeoff or not given typical usage, e.g. how likely it is the first check will succeed or fail.

Reply:
|   Method | Length |     Mean |    Error | Ratio |
|--------- |------- |---------:|---------:|------:|
|   2 \|\| | 100000 | 87.79 us | 0.290 us |  1.00 |
|     2 \| | 100000 | 87.81 us | 0.369 us |  1.00 |
|          |        |          |          |       |
|   3 \|\| | 100000 | 110.8 us |  0.66 us |  1.00 |
|     3 \| | 100000 | 109.7 us |  0.48 us |  0.99 |
|          |        |          |          |       |
|   4 \|\| | 100000 | 109.9 us |  0.52 us |  1.00 |
|     4 \| | 100000 | 181.2 us |  0.53 us |  1.65 |

Codegen for (`|` is on the right)
- 2 values: https://www.diffchecker.com/3ytgU658
- 4 values: https://www.diffchecker.com/LfsR8ge8

Reply:
These are all for cases where the condition is never hit

Reply:
We could also experiment with the sort of optimizations like your (`[x, x + 32)`) in the future here

Reply:
> These are all for cases where the condition is never hit

Yeah, it'll depend on the data and how likely the branch predictor is to get it right.  If the condition is never hit, the branch predictor is basically always going to be right, both clauses will always execute, and they should effectively be identical, which is what your data shows (at least for 2 and 3... I'm surprised 4 falls off a cliff).

If I instead change the data to alternate between matching the first char and not matching anything, the data looks different, at least on my machine:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

[MemoryDiagnoser]
public partial class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    private char _c1, _c2;

    private char[] _values = new char[100000];

    [GlobalSetup]
    public void Setup()
    {
        for (int i = 0; i < _values.Length; i++)
        {
            _values[i] = (char)(i % 2);
        }
        _c1 = (char)0;
        _c2 = 'c';
    }

    [Benchmark(Baseline = true)]
    public int Count_Logical()
    {
        int count = 0;
        foreach (char c in _values)
        {
            if ((_c1 == c) || (_c2 == c)) count++;
        }
        return count;
    }

    [Benchmark]
    public int Count_Bitwise()
    {
        int count = 0;
        foreach (char c in _values)
        {
            if ((_c1 == c) | (_c2 == c)) count++;
        }
        return count;
    }
}
```

|        Method |      Mean |    Error |   StdDev | Ratio |
|-------------- |----------:|---------:|---------:|------:|
| Count_Logical | 118.12 us | 0.337 us | 0.315 us |  1.00 |
| Count_Bitwise |  77.48 us | 0.423 us | 0.353 us |  0.66 |

Reply:
Interesting... do you think it's worth changing to `|` (at least for 2 & 3) in that case?